### ISSUE TITLE
No error raise for wrong type of result

### ISSUE THREAD

**Issue by ZhiyuanChen** [microsoft/nni#3641]
**Describe the issue**:

When parsing value in `torch.tensor` with `dtype=torch.float` to `nni.report_intermediate_result` and `nni.report_final_result`, there is no Exception raised and no result recorded. 

It is desired to raise a `ValueError`, or auto-convert to python float (raise a `ValueError` when cannot be converted). 

**Environment**:
- NNI version:
- Training service (local|remote|pai|aml|etc):
- Client OS:
- Server OS (for remote mode only):
- Python version:
- PyTorch/TensorFlow version:
- Is conda/virtualenv/venv used?:
- Is running in Docker?:

**Configuration**:
 - Experiment config (remember to remove secrets!):
 - Search space:

**Log message**:
 - nnimanager.log:
 - dispatcher.log:
 - nnictl stdout and stderr:
 

**How to reproduce it?**:

**Comment by ultmaster** [comment]
Excellent! Could you submit a quick fix PR to check the type of final result?

**Comment by ZhiyuanChen** [comment]
> Excellent! Could you submit a quick fix PR to check the type of final result?

Sure thing

**Comment by ZhiyuanChen** [comment]
Hi @ultmaster
Can I confirm the only possible types allowed for `metric` are `float` and `dict`? 
also, why not just use json.dumps here which gives the same result with proper l
